### PR TITLE
Add a Transit Gateway to the Shared Services account

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

In this pull request I add a Transit Gateway to the Shared Services account.

## 💭 Motivation and Context

We require a Transit Gateway to allow inter-VPC communication.

## 🧪 Testing

I boldly applied these changes to production.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by felddy — APPROVED** [review]
>Testing: I boldly applied these changes to production.

I like the cut of your jib Sir! ⛵️

File: private_routing.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# uses its own NAT gateway.
+resource "aws_route_table" "private_route_tables" {
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  tags   = var.tags
+  vpc_id = aws_vpc.the_vpc.id
+}
+
+# Route all non-local COOL (outside this VPC but inside the COOL)
+# traffic this VPC through the transit gateway.
```
Comment (dav3r) [inline private_routing.tf]: I think you meant to delete "this VPC" from this comment?

File: public_routing.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# -------------------------------------------------------------------------------
+
+# Default route table (used by public subnets)
+resource "aws_default_route_table" "public" {
+  default_route_table_id = aws_vpc.the_vpc.default_route_table_id
+  tags                   = var.tags
+}
+
+# Route all non-local COOL (outside this VPC but inside the COOL)
+# traffic this VPC through the transit gateway
```
Comment (dav3r) [inline public_routing.tf]: I think you meant to delete "this VPC" from this comment?

**Review by dav3r — APPROVED** [review]
Delicious! 🔄 
Please note the small comment typos I mentioned.

File: public_routing.tf (hunk 2 above)
Comment (jsf9k) [reply public_routing.tf]: Fixed in commit 9c5e061.

File: private_routing.tf (hunk 1 above)
Comment (jsf9k) [reply private_routing.tf]: Fixed in commit 9c5e061.

**Pull request merged**